.github/workflows/ci.yml

```
@@ -24,3 +25,23 @@ jobs:
     - name: Run tests
       run: bundle exec rspec
 
+  publish-gem:
+    if: github.ref_type == 'tag'
```

I think because of this conditional, the publish step might still run even if tests don't pass - [see this section in the docs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-not-requiring-successful-dependent-jobs).

I'm pretty sure we can add a check for the result of the previous step:

```suggestion
    if: github.ref_type == 'tag' && needs.test.result == 'success'
```

---

Hey. The doc is a bit confusing, and my first thought was the same. But looks like you need to use specifically [Conditional Expression](https://docs.github.com/en/actions/learn-github-actions/expressions#status-check-functions) to overwrite default behavior, not any `if` condition.

> If a job fails, all jobs that need it are skipped unless the jobs use a conditional expression that causes the job to continue.
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds

> You can use the following status check functions as expressions in if conditionals. A default status check of success() is applied unless you include one of these functions.
https://docs.github.com/en/actions/learn-github-actions/expressions#status-check-functions

I also checked on my repository, you can see that image publishing is skipped because lint job failed here: https://github.com/flexoid/pivotal-slack-preview/actions/runs/1735584038
https://github.com/flexoid/pivotal-slack-preview/blob/main/.github/workflows/go.yml

---

Ah that is confusing! Thanks for clarifying! 🙂